src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -2,181 +2,74 @@
+        line.addSection(request.toSection(upStation, downStation));
     }
 
+    @Transactional
     public void removeLineStation(Long lineId, Long stationId) {
         Line line = findLineById(lineId);
-        Station station = stationService.findStationById(stationId);
-        if (line.getSections().size() <= 1) {
-            throw new RuntimeException();
-        }
-
-        Optional<Section> upLineStation = line.getSections().stream()
-                .filter(it -> it.getUpStation() == station)
-                .findFirst();
-        Optional<Section> downLineStation = line.getSections().stream()
-                .filter(it -> it.getDownStation() == station)
-                .findFirst();
-
-        if (upLineStation.isPresent() && downLineStation.isPresent()) {
-            Station newUpStation = downLineStation.get().getUpStation();
-            Station newDownStation = upLineStation.get().getDownStation();
-            int newDistance = upLineStation.get().getDistance() + downLineStation.get().getDistance();
-            line.getSections().add(new Section(line, newUpStation, newDownStation, newDistance));
-        }
-
-        upLineStation.ifPresent(it -> line.getSections().remove(it));
-        downLineStation.ifPresent(it -> line.getSections().remove(it));
+        Station station = stationService.findById(stationId);
+        line.removeSection(station);
     }
 
-
-    public List<Station> getStations(Line line) {
-        if (line.getSections().isEmpty()) {
-            return Arrays.asList();
-        }
-
-        List<Station> stations = new ArrayList<>();
-        Station downStation = findUpStation(line);
-        stations.add(downStation);
-
-        while (downStation != null) {
-            Station finalDownStation = downStation;
-            Optional<Section> nextLineStation = line.getSections().stream()
-                    .filter(it -> it.getUpStation() == finalDownStation)
-                    .findFirst();
-            if (!nextLineStation.isPresent()) {
-                break;
-            }
-            downStation = nextLineStation.get().getDownStation();
-            stations.add(downStation);
-        }
-
-        return stations;
-    }
-
-    private Station findUpStation(Line line) {
-        Station downStation = line.getSections().get(0).getUpStation();
-        while (downStation != null) {
-            Station finalDownStation = downStation;
-            Optional<Section> nextLineStation = line.getSections().stream()
-                    .filter(it -> it.getDownStation() == finalDownStation)
-                    .findFirst();
-            if (!nextLineStation.isPresent()) {
-                break;
-            }
-            downStation = nextLineStation.get().getUpStation();
-        }
-
-        return downStation;
+    private Line findLineById(Long id) {
+        return lineRepository.findById(id).orElseThrow(RuntimeException::new);
```

Comment:
하나의 라인에 하나의 점을 사용해주시면 가독성이 올라갑니다 :)

```suggestion
        return lineRepository.findById(id)
                .orElseThrow(RuntimeException::new);
```


그리고 RuntimeException은 예외의 super class의 역할을 하고 있기 때문에 굉장히 추상적인 예외입니다.
해당 부분에 어울리는 custom exception을 만들어보는건 어떨까요 ? 
custom exception에 관련된 흥미로운 링크를 남겨놓겠습니다 !
[https://tecoble.techcourse.co.kr/post/2020-08-17-custom-exception/](https://tecoble.techcourse.co.kr/post/2020-08-17-custom-exception/)

추가적으로 에외가 발생했을때 처리하는 로직도 함께 구현해보면 좋을 것 같아요 😄

Reply:
넵 수정하였습니다. 커스텀 예외도 만들어보았습니다~! 
예외를 만들 때 첨부해 주신 자료 참고해서 목적에 맞게 사용해 보도록 하겠습니다.